pkg/system/rm.go

```diff
@@ -78,3 +78,31 @@ func EnsureRemoveAll(dir string) error {
 		time.Sleep(100 * time.Millisecond)
 	}
 }
+
+// AtomicRemoveAll performs an atomic remove using `EnsureRemoveAll`
+// During removal the passed in dir will not be accessible.
+func AtomicRemoveAll(dir string) error {
```

Comment:
Heads up, I really don't like this abstraction even though it would clean up some code elsewhere as well, I'm not sure it's worth it.

Reply:
Should we un-export it, and move it to plugin for now?

Reply:
@cpuguy83 why is that ? 🤔 👼

Reply:
It's generalizing something that's not that easy to generalize.